docs: Add docs for format strings (#1083)
starship/starship: docs/config/README.md

**Module**: A component in the prompt giving information based on contextual information from your OS. For example, the "nodejs" module shows the version of NodeJS that is currently installed on your computer, if your current directory is a NodeJS project.

**Variable**: Smaller sub-components that contains information provided by the module. For example, the "version" segment in the "nodejs" module contains the current version of NodeJS.

By convention, most modules have a prefix of default terminal color (e.g. `via ` in "nodejs") and an empty space as a suffix.

### Format Strings

Format strings are the format that a module prints all its variables with.
Most modules have an entry called `format` that configures the display format of the module.
You can use texts, variables and text groups in a format string.

#### Variable

A variable contains a `$` symbol followed by the name of the variable.

> The name of a variable only contains alphabetics, numerics and `_`.

Reviewer: ```suggestion
The name of a variable only contains letters, numbers and `_`.
```

For example:

- `$version` is a format string with a variable named `version`.
- `$git_branch$git_commit` is a format string with two variables named `git_branch` and `git_commit`.
- `$git_branch $git_commit` has the two variables separated with a space.

#### Text Group

> A text group is combined with two different parts.

Reviewer: ```suggestion
A text group is made up of two different parts.
```

The first part, which is enclosed in a `[]`, is a [format string](#format-strings).
You can add texts, variables, or even nested text groups in it.

> In the second part, a [style string](#style-strings) should be provided to paint the first part with.

Reviewer: ```suggestion
In the second part, which is enclosed in a `()`, is a [style string](#style-strings). This can be used style the first part.
```

Note that what styling looks like will be controlled by your terminal emulator. For example, some terminal emulators will brighten the colors instead of bolding text, and some color themes use the same values for the normal and bright colors. Also, to get italic text, your terminal must support italics.

#### Escapable characters

The following symbols have special usage in a format string.
If you want to print the following symbols, you have to escape them with a backslash (`\`).

- `$`
- `\`
- `[`
- `]`
- `(`
- `)`

Note that `toml` has [its own escape syntax](https://github.com/toml-lang/toml#user-content-string).
It is recommended to use a literal string (`''`) in your config.
If you want to use a basic string (`""`), pay attention to escape the backslash `\`.

> For example, when you want to print a `$` symbol in a new line, the following configs for `format` are equivalent:

Reviewer: ```suggestion
For example, when you want to print a `$` symbol on a new line, the following configs for `format` are equivalent:
```
